--- Makefile	2014-08-19 14:30:39.187807254 +0000
+++ Makefile.patched	2014-08-19 14:30:36.283807287 +0000
@@ -452,7 +452,7 @@
 LD = /usr/bin/ld
 LDFLAGS = -L/usr/local/hyperdex/lib -fPIC -fpic
 LIBOBJS = 
-LIBS = -lpopt 
+LIBS = /usr/local/hyperdex/lib/libpopt.a
 LIBTOOL = $(SHELL) $(top_builddir)/libtool
 LIPO = 
 LN_S = ln -s
@@ -581,7 +581,7 @@
 	daemon/heal_next.cc daemon/main.cc daemon/object_manager.cc \
 	daemon/settings.cc daemon/snapshot.cc
 replicant_daemon_LDADD = $(HYPERLEVELDB_LIBS) $(BUSYBEE_LIBS) \
-	$(E_LIBS) libreplicant-state-machine.la -lglog -ldl -lpthread
+	$(E_LIBS) libreplicant-state-machine.la /usr/local/hyperdex/lib/libglog.a -ldl -lpthread
 libreplicant_state_machine_la_SOURCES = daemon/replicant_state_machine_context.cc
 libreplicant_state_machine_la_LDFLAGS = -version-info 1:0:0
 libreplicant_la_SOURCES = common/bootstrap.cc common/chain_node.cc \
@@ -595,23 +595,23 @@
 replicant_SOURCES = replicant.cc
 replicant_CPPFLAGS = -DREPLICANT_EXEC_DIR="\"$(replicantexecdir)\"" $(AM_CPPFLAGS) $(CPPFLAGS)
 replicant_block_SOURCES = tools/block.cc
-replicant_block_LDADD = libreplicant.la
+replicant_block_LDADD = -static-libtool-libs libreplicant.la $(BUSYBEE_LIBS) $(E_LIBS)
 replicant_cli_SOURCES = tools/cli.cc
-replicant_cli_LDADD = libreplicant.la
+replicant_cli_LDADD = -static-libtool-libs libreplicant.la $(BUSYBEE_LIBS) $(E_LIBS)
 replicant_new_object_SOURCES = tools/new-object.cc
-replicant_new_object_LDADD = libreplicant.la
+replicant_new_object_LDADD = -static-libtool-libs libreplicant.la $(BUSYBEE_LIBS) $(E_LIBS)
 replicant_del_object_SOURCES = tools/del-object.cc
-replicant_del_object_LDADD = libreplicant.la
+replicant_del_object_LDADD = -static-libtool-libs libreplicant.la $(BUSYBEE_LIBS) $(E_LIBS)
 replicant_backup_object_SOURCES = tools/backup-object.cc
-replicant_backup_object_LDADD = libreplicant.la
+replicant_backup_object_LDADD = -static-libtool-libs libreplicant.la $(BUSYBEE_LIBS) $(E_LIBS)
 replicant_restore_object_SOURCES = tools/restore-object.cc
-replicant_restore_object_LDADD = libreplicant.la
+replicant_restore_object_LDADD = -static-libtool-libs libreplicant.la $(BUSYBEE_LIBS) $(E_LIBS)
 replicant_integrity_check_SOURCES = tools/integrity-check.cc common/chain_node.cc common/configuration.cc common/packing.cc common/response_returncode.cc daemon/configuration_manager.cc daemon/fact_store.cc
-replicant_integrity_check_LDADD = $(HYPERLEVELDB_LIBS) $(E_LIBS) -lglog
+replicant_integrity_check_LDADD = $(HYPERLEVELDB_LIBS) $(E_LIBS) /usr/local/hyperdex/lib/libglog.a
 replicant_debug_dump_SOURCES = tools/debug-dump.cc common/chain_node.cc common/configuration.cc common/packing.cc common/response_returncode.cc daemon/configuration_manager.cc daemon/fact_store.cc
-replicant_debug_dump_LDADD = $(HYPERLEVELDB_LIBS) $(E_LIBS) -lglog
+replicant_debug_dump_LDADD = $(HYPERLEVELDB_LIBS) $(E_LIBS) /usr/local/hyperdex/lib/libglog.a
 replicant_list_servers_SOURCES = tools/list-servers.cc
-replicant_list_servers_LDADD = libreplicant.la
+replicant_list_servers_LDADD = -static-libtool-libs libreplicant.la $(BUSYBEE_LIBS) $(E_LIBS)
 libreplicant_example_alarm_la_SOURCES = examples/alarm.c
 libreplicant_example_alarm_la_CPPFLAGS = -I$(abs_top_srcdir)/daemon/ $(AM_CPPFLAGS) $(CPPFLAGS)
 libreplicant_example_condition_la_SOURCES = examples/condition.c
